ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java

```diff
@@ -281,30 +303,39 @@ private NettyPerfClient(String host, int port, String path, int concurrency, Lon
       this.totalSize = 0;
       chunk = null;
     }
-    sslFactory = sslPropsFilePath != null ? new SSLFactory(
-        new SSLConfig(new VerifiableProperties(Utils.loadProps(sslPropsFilePath)))) : null;
+    if (sslPropsFilePath != null) {
+      SSLConfig sslConfig = new SSLConfig(new VerifiableProperties(Utils.loadProps(sslPropsFilePath)));
+      if (useOpenSsl) {
+        if (SslContext.defaultClientProvider() == SslProvider.JDK) {
+          throw new IllegalStateException("OpenSsl not supported. netty-tcnative might not be loaded correctly");
+        }
+        logger.info("Using this buffer pool: " + ByteBufAllocator.DEFAULT);
+        sslFactory = new NettySslFactory(sslConfig);
+      } else {
+        sslFactory = new JdkSslFactory(sslConfig);
```

Comment:
why not use `SSLFactory.getNewInstance()` here?

Reply:
I will do that. Then you would be able to change it through the properties file instead of a command line arg.